rotkehlchen/exchanges/gemini.py

```diff
@@ -554,9 +594,79 @@ def query_online_deposits_withdrawals(
                 continue
 
             movements.append(movement)
-
+            with self.db.user_write() as write_cursor:
+                if len(history_events) != 0:
+                    db = DBHistoryEvents(self.db)
+                    db.add_history_events(write_cursor=write_cursor, history=history_events)
+                self.db.set_dynamic_cache(
+                    write_cursor=write_cursor,
+                    name=DBCacheDynamic.LAST_QUERY_TS,
+                    value=ts_now(),
+                    location=self.location.serialize(),
+                    location_name=self.name,
+                )
         return movements
 
+    def _deserialize_history_event(
+            self,
+            raw_data: dict[str, Any],
+            timestamp: Timestamp,
+            asset: AssetWithOracles,
+            event_type: AssetMovementCategory,
+        ) -> HistoryEvent | None:
+        """Processes a single transaction from coinbase and deserializes it
+
+        Can log error/warning and return None if something went wrong at deserialization
+        """
+        try:
+            if raw_data.get('status').lower() != 'complete':
+                return None
+            if raw_data.get('type').lower() != 'reward':
+                return None
+
+            amount = deserialize_asset_amount(raw_data.get('amount'))
+            event_subtype = HistoryEventSubType.REWARD
+            if raw_data.get('method') == 'CreditCard':
+                notes = 'Gemini Credit Card Reward'
+            else:
+                notes = ''
+            return HistoryEvent(
+                event_identifier=f'{GEMINI_EVENTS_PREFIX}{raw_data["eid"]!s}',
+                sequence_index=0,
+                timestamp=timestamp,
+                location=Location.GEMINI,
+                event_type=event_type,
+                event_subtype=event_subtype,
+                asset=asset,
+                balance=Balance(amount=amount, usd_value=ZERO),
```

Comment:
Default value is ZERO

```suggestion
                balance=Balance(amount=amount),
```